source/agora/node/Registry.d

```
@@ -680,6 +687,194 @@ private struct ZoneData
     /// The content of the zone
     public TypedPayload[PublicKey] map;
 
+    private string type_table_name;
+    private ManagedDatabase db;
+    private Logger log;
+
+    public this(string type_table_name, ManagedDatabase db, Logger logger)
+    {
+        this.type_table_name = type_table_name;
+        this.db = db;
+        this.log = logger;
+
+        string query_signature = format("CREATE TABLE IF NOT EXISTS registry_%s_signature " ~
+            "(pubkey TEXT, signature TEXT NOT NULL, sequence INTEGER NOT NULL, " ~
+            "PRIMARY KEY(pubkey))", this.type_table_name);
+        
+        string query_addresses = format("CREATE TABLE IF NOT EXISTS registry_%s_addresses " ~
+            "(pubkey TEXT, address TEXT NOT NULL, type INTEGER NOT NULL, " ~
+            "FOREIGN KEY(pubkey) REFERENCES registry_%s_signature(pubkey) ON DELETE CASCADE, " ~
+            "PRIMARY KEY(pubkey, address))", this.type_table_name, this.type_table_name);
+
+        try
+        {
+            this.db.execute(query_signature);
+            this.db.execute(query_addresses);
+        }
+        catch (Exception ex)
+        {
+            this.log.warn("ManagedDatabase operation error: {}", ex.msg);
+        }
+    }
+
+    /***************************************************************************
+
+         Returns:
+           Total number of name registries
+
+    ***************************************************************************/
+    public ulong getRegistryCount () @trusted
+    {
+        try
+        {
+            string query = format("SELECT COUNT(*) FROM registry_%s_signature", this.type_table_name);
+            auto results = this.db.execute(query);
+            if (!results.empty)
+            {
+                return results.oneValue!(ulong);
+            }
+        }
+        catch (Exception ex)
+        {
+            this.log.warn("ManagedDatabase operation error: {}", ex.msg);
+        }
+
+        return 0;
```

It seems a bit off to me to swallow the Exception and return an in-band value. I know a lot of place swallow exceptions, but it's quite an anti pattern, I'd say let it propagate.